### ISSUE TITLE
Databricks magic dialect linting LT01 issues

### ISSUE THREAD

**Issue by malthe** [sqlfluff/sqlfluff#8376]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I get a linting error when using the feature proposed in https://github.com/sqlfluff/sqlfluff/issues/7877 on even the most simple notebook.

> LT01 Expected single whitespace between magic start and magic line. [layout.spacing]

For example, we can try linting the included test fixture:

```bash
$ sqlfluff lint -d databricks test/fixtures/dialects/databricks/magic_line.sql
== [test/fixtures/dialects/databricks/magic_line.sql] FAIL                                                                                                                                                   
L:   3 | P:   1 | LT01 | Expected single whitespace between magic start and magic
                       | line. [layout.spacing]
L:  14 | P:   1 | LT01 | Expected single whitespace between magic start and magic
                       | line. [layout.spacing]
L:  24 | P:   1 | LT01 | Expected single whitespace between magic start and magic
                       | line. [layout.spacing]
```
The LT01 here makes no sense.

### Expected Behaviour

No linting error.

### Observed Behaviour

We get an LT01.

### How to reproduce

```bash
$ sqlfluff lint -d databricks test/fixtures/dialects/databricks/magic_line.sql
```

### Dialect

Databricks

### Version

sqlfluff, version 4.3.0

### Configuration

# Empty

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by malthe** [comment]
@peterbud this happens with the changes in https://github.com/sqlfluff/sqlfluff/pull/8011.

**Comment by peterbud** [comment]
Hi @malthe - thanks. 

The original PR was about making MAGIC lines to parse. It' wasn't about linting, and as I see newline is consumed inside `magic_start`, the layout engine sees those two code segments as directly adjacent, with no standalone whitespace/newline segment between them.

Definitely needs to be fixed.

**Comment by malthe** [comment]
@peterbud see also https://github.com/sqlfluff/sqlfluff/pull/8377 (opened just now by me).

**Comment by peterbud** [comment]
> [@peterbud](https://github.com/peterbud) see also [#8377](https://github.com/sqlfluff/sqlfluff/pull/8377) (opened just now by me).

Hi @malthe  many thanks for opening that, I have seen it. As I see that is NOT fixing this linting problem, as the newline is still part of the `magic_start` token, so LT01 will still report it.

Look at the PR above, I have fixed these issues in that.
